### PR TITLE
Refactor analysis points to include more precise error data

### LINKED ISSUE #5687: Investigate whether it's possible to tell if an analysis error is because it's 10k+, or because of another reason
```
jlast — Today at 3:08 PM
@Jaril (he/him) i think we're looking into bugs with the 10k hits
jlast — Today at 3:08 PM
Oh interesting InvalidObject is what we show when we fail to understand the object. CC @hbenl (he/him)
@Andarist is it possible to share this replay with us with comments at these spots?
Andarist — Today at 3:11 PM
invalid object stuff here
https://app.replay.io/recording/composed-conditional-type-expands-into-union-of-its-its-branches--3362459f-d68d-40af-997b-7f407643f642
10k+ hits are reported almost everywhere, can be seen on the same line
Jaril (he/him) — Today at 3:13 PM
I think we infer that something has 10k+ hits when the analysis returns an error. Which it might have, but it's also possible that any other number of other errors could have happened, and it's just being masked by the 10k+ label
Andarist — Today at 3:14 PM
might be it, i've refreshed and i don't experience this right now
aaaaand i have it again 😄
jlast — Today at 3:16 PM
@Jaril (he/him) shouldnt we ahve an error code for the 10K to differentiate
Andarist — Today at 3:16 PM
actually, it behaves slightly differently now - i dont see it everywhere now, but i have some lines on which this appears and im pretty sure that this info is invalid on those lines
oh, sorry - i was looking at it wrong, this time this actually might be reported correctly here as its a super generic helper, i was just htinking about its parent call which isnt called as much and the info is OK there 
```

### PR DESCRIPTION
**Pull request by jaril**
Fix #5687.

### CODE REVIEW

File: src/ui/state/app.ts [hunk 1]
```diff
@@ -100,8 +100,14 @@ export interface AppState {
...
 
-export interface AnalysisPoints {
-  [key: string]: PointDescription[] | "error";
+export type AnalysisPoints = Record<string, AnalysisPayload>;
+export type AnalysisPayload = {
+  data: PointDescription[];
+  error: AnalysisError | null;
+};
+export enum AnalysisError {
+  HellaHits = "hella-hits",
```
Comment (jcmorrow) [inline src/ui/state/app.ts:109]: 🤨

File: src/ui/state/app.ts [hunk 2]
```diff
@@ -126,3 +132,7 @@ export interface Canvas {
   top: number;
   width: number;
 }
+
+export enum ProtocolError {
+  TooManyPoints = 55,
```
Comment (jcmorrow) [inline src/ui/state/app.ts:137]: We don't export this type from protocol huh?

File: src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointNavigation.js [hunk 3]
```diff
@@ -58,7 +58,7 @@ function BreakpointNavigation({
       {!isEmpty ? (
         <BreakpointNavigationCommands prev={prev} next={next} navigateToPoint={navigateToPoint} />
       ) : null}
-      {analysisPoints !== "error" ? (
+      {analysisPoints.error ? (
```
Comment (hbenl) [inline src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointNavigation.js:61]: Should be `!analysisPoints.error`

File: src/protocol/logpoint.ts [hunk 4]
```diff
@@ -253,8 +250,8 @@ async function setMultiSourceLogpoint(
   if (showInConsole && primitiveFronts) {
     const points = getAnalysisPointsForLocation(store.getState(), locations[0], condition);
     if (points) {
-      if (points !== "error") {
-        showPrimitiveLogpoints(logGroupId, points, primitiveFronts);
+      if (points.error) {
```
Comment (hbenl) [inline src/protocol/logpoint.ts:253]: should be `!points.error`

File: src/ui/reducers/app.ts [hunk 5]
```diff
@@ -237,28 +264,16 @@ export const getAnalysisPointsForLocation = (
...
   const location = getHoveredLineNumberLocation(state);
-  const points = getAnalysisPointsForLocation(state, location);
-  const focusRegion = getFocusRegion(state);
-
-  if (focusRegion && points && points !== "error") {
-    return points.filter(p => isInTrimSpan(p.time, focusRegion));
-  }
-
-  return points;
+  return getAnalysisPointsForLocation(state, location);
```
Comment (hbenl) [inline src/ui/reducers/app.ts:276]: no `getPointsInTrimSpan()`?

**Review by hbenl — APPROVED** [review]
Found 2 missing negations and a possible oversight. Otherwise LGTM.

File: src/ui/reducers/app.ts (hunk 5 above)
Comment (jaril) [reply src/ui/reducers/app.ts:276]: It's applied inside `getAnalysisPointsForLocation` so no need here. Let me know if I missed something!

**Pull request merged**